### ISSUE TITLE
SoX is not executable

### ISSUE THREAD

**Issue by delucis** [ArtskydJ/sox-bin#6]
On OS X, if I `npm i sox-bin` and try using it as described in the README, I receive a “Permission denied” error.

To make it work, I have to make it executable after installing. For example:

```js
const sox = require('sox-bin')
const cp = require('child_process')

const command = `'chmod +x ${sox}'`

cp.exec(command, err => {
  if (err) throw err
})
```

Is this the expected behaviour?

**Comment by ArtskydJ** [comment]
I was under the impression that this https://github.com/ArtskydJ/sox-bin/commit/210e1c18eade59079371d83c97f9733be8c7d1e1 fixed permission issues on Macintosh, but I guess it did not.

Just curious, if you install from github rather than npm, are the permissions correct? (I don't have a mac on which to test :frowning_face:)

**Comment by delucis** [comment]
Hi @ArtskydJ, installing from GitHub — `npm i -S ArtskydJ/sox-bin` — does indeed work. So maybe `npm` does something to permissions for the files it serves?

**Comment by ArtskydJ** [comment]
Is your npm client up-to-date? There's a decent chance that they set everything to non-executable for security reasons... IDK

If you want, you could make a PR to the readme with a `chmod ...` command... Either way, I'll leave this issue open until the README is fixed, or the issue is fixed.

Maybe there's a way to do [`fs.chmod`](https://nodejs.org/api/fs.html#fs_fs_chmod_path_mode_callback) if it detects that it is necessary.

**Comment by delucis** [comment]
I was using npm v5.10.0, so tried updating to v6.1.0, but the issue is the same.

I wonder if [an `install` script](https://docs.npmjs.com/misc/scripts) could be added that checks whether the binary is executable (possible using [just `fs.access`](https://stackoverflow.com/a/41929624) or a package like [`bin-check`](https://github.com/kevva/bin-check)) and runs `fs.chmod` if necessary? The NPM docs [recommend not using `install` scripts](https://docs.npmjs.com/misc/scripts#best-practices), but this seems like a legitimate use.

I’m a little hesitant because I can only test on a Mac, but can try a PR if you’re interested.

**Comment by ArtskydJ** [comment]
I made [this branch](https://github.com/ArtskydJ/sox-bin/tree/npm-install-fix-permissions)... I have no idea if it works... I guess there's not really a great way to test it unless I publish it to NPM.

If you want, you could do a fresh `npm install sox-bin`, then copy the new `install.js` file to that directory, and run `node install.js`, and then see if the permissions are good.

Let me know if you do test it, or if you don't plan to, and I can see what I can do. I might enable the install script for all non-win32 platforms, idk...

**Comment by delucis** [comment]
Created a repo to run some tests against that branch on Travis-CI and here are the results: https://travis-ci.com/delucis/soxbintest/builds/76135365

Looks like that branch fixes the problem for OS X. The Linux builds are failing for some other reason though (as they are on the main branch already).

**Comment by ArtskydJ** [comment]
Oh, I had forgotten about installing from gh. I wonder if it handles permissions the same way...

Anyway, I merged the permissions fix. I plan to deploy it to npm after work. (8+ hours from now)

**Comment by delucis** [comment]
Oh right, installing from GitHub worked before too. One option could be to publish a prerelease to NPM behind a `next` tag. That could then be tested to be sure it works before making it a full release.

**Comment by ArtskydJ** [comment]
Good idea, I added it with a `beta` tag.

**Comment by delucis** [comment]
Great! Looks like it’s still working 🎉
https://travis-ci.com/delucis/soxbintest/builds/76256742

Thanks for working on this 😄

**Comment by ArtskydJ** [comment]
Published as 1.0.2
